src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -5167,10 +5284,10 @@ internal AutomaticVarSaver(Compiler compiler, VariablePath autoVarPath, int auto
                 _automaticVar = automaticVar;
             }
 
-            internal IEnumerable<ParameterExpression> GetTemps()
+            internal IReadOnlyList<ParameterExpression> GetTemps()
             {
                 Diagnostics.Assert(_oldValue != null, "caller to only call GetTemps after calling SaveAutomaticVar");
-                yield return _oldValue;
+                return new ParameterExpression[] { _oldValue };
```

Comment:
It seems we do extra allocation with the change.

Reply:
It doesn't require an enumerator method, though. 🤷 

The other option would be to just... not use a collection type at all here, since we know for sure this is never going to be a collection. 

That'll require some additional refactoring but should be workable.

Reply:
Forgot to update this thread, but I ended up doing this.